tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+'''
+Utility script to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import ipaddress
+import json
+import pathlib
+from os.path import dirname, abspath
+import requests
+import subprocess
+import sys
+
+
+def query_prometheus_api(ip, port, route=None):
+    """Read the Prometheus route contents into memory."""
+    if not route:
+        sys.stderr.write("Expected endpoint route but got None")
+        sys.exit(1)
+
+    endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(ip, port, route)
+    try:
+        response = requests.get(url=endpoint, verify=False)
+    except requests.exceptions.ConnectionError as exc:
+        sys.stderr.write(
+            "Failed to reach endpoint {}: {!s}".format(route, exc)
+        )
+        sys.exit(1)
+
+    if response.status_code != 200:
+        sys.stderr.write(
+            "Expected status code <200> got {}".format(response.status_code)
+        )
+        sys.exit(1)
+
+    return response.json()
+
+
+def write_to_file(content, rule_name):
+    filename = '{}.json'.format(rule_name)
+    filepath = pathlib.Path(__file__).parent/filename
+    with open('{}.json'.format(filepath), 'w', encoding='utf-8') as f:
+        f.write(json.dumps(content, indent=4, sort_keys=True))
```

Comment:
Minor: consider using `json.dump` instead of `write` + `json.dumps`

Reply:
Done